sentry-android-core/src/main/java/io/sentry/android/core/NetworkBreadcrumbsIntegration.java

```diff
@@ -63,40 +63,71 @@ public void register(final @NotNull IHub hub, final @NotNull SentryOptions optio
         "NetworkBreadcrumbsIntegration enabled: %s",
         androidOptions.isEnableNetworkEventBreadcrumbs());
 
+    this.options = options;
+
     if (androidOptions.isEnableNetworkEventBreadcrumbs()) {
 
       // The specific error is logged in the ConnectivityChecker method
-      if (buildInfoProvider.getSdkInfoVersion() < Build.VERSION_CODES.LOLLIPOP) {
-        networkCallback = null;
-        logger.log(SentryLevel.DEBUG, "NetworkBreadcrumbsIntegration requires Android 5+");
+      if (buildInfoProvider.getSdkInfoVersion() < Build.VERSION_CODES.N) {
+        logger.log(SentryLevel.DEBUG, "NetworkCallbacks need Android N+.");
         return;
       }
 
       networkCallback =
           new NetworkBreadcrumbsNetworkCallback(hub, buildInfoProvider, options.getDateProvider());
-      final boolean registered =
-          AndroidConnectionStatusProvider.registerNetworkCallback(
-              context, logger, buildInfoProvider, networkCallback);
 
-      // The specific error is logged in the ConnectivityChecker method
-      if (!registered) {
-        networkCallback = null;
-        logger.log(SentryLevel.DEBUG, "NetworkBreadcrumbsIntegration not installed.");
-        return;
+      try {
+        options
+            .getExecutorService()
+            .submit(
+                new Runnable() {
+                  @Override
+                  public void run() {
+                    // in case integration is closed before the task is executed, simply return
+                    final @Nullable NetworkBreadcrumbsNetworkCallback callback = networkCallback;
+                    if (isClosed || callback == null) {
```

Comment:
was thinking if we should synchronize here, but since we use the same thread (executorService), I guess it should be fine, right?